test/functional/editor/jump_spec.lua

```diff
@@ -182,6 +182,105 @@ describe("jumpoptions=stack behaves like 'tagstack'", function()
   end)
 end)
 
+describe('buffer deletion', function()
+  local base_file = 'Xtest-functional-buffer-deletion'
+  local file1 = base_file..'1'
+  local file2 = base_file..'2'
+  local file3 = base_file..'3'
+  local base_content = 'text'
+  local content1 = base_content..'1'
+  local content2 = base_content..'2'
+  local content3 = base_content..'3'
+
+  before_each(function()
+    clear()
+    command('clearjumps')
+
+    write_file(file1, content1, false, false)
+    write_file(file2, content2, false, false)
+    write_file(file3, content3, false, false)
+
+    command("edit " .. file1)
+    command("edit " .. file2)
+    command("edit " .. file3)
+  end)
+
+  it('deletes jump list entries when the current buffer is deleted', function()
+    command("edit " .. file1)
+
+    eq(   ''
+       .. ' jump line  col file/text\n'
+       .. '   3     1    0 '..content1..'\n'
+       .. '   2     1    0 '..file2..'\n'
+       .. '   1     1    0 '..file3..'\n'
+       .. '>',
+       exec_capture('jumps'))
+
+    command('bwipeout')
+
+    eq(   ''
+       .. ' jump line  col file/text\n'
+       .. '   1     1    0 '..file2..'\n'
+       .. '>  0     1    0 '..content3,
+       exec_capture('jumps'))
+  end)
+
+  it('deletes jump list entries when another buffer is deleted', function()
+    eq(   ''
+       .. ' jump line  col file/text\n'
+       .. '   2     1    0 '..file1..'\n'
+       .. '   1     1    0 '..file2..'\n'
+       .. '>',
+       exec_capture('jumps'))
+
+    command('bwipeout '..file2)
+
+    eq(   ''
+       .. ' jump line  col file/text\n'
+       .. '   1     1    0 '..file1..'\n'
+       .. '>',
+       exec_capture('jumps'))
+  end)
+
+  it('sets the correct jump index when the current buffer is deleted', function()
+    feed('<C-O>')
+
+    eq(   ''
+       .. ' jump line  col file/text\n'
+       .. '   1     1    0 '..file1..'\n'
+       .. '>  0     1    0 '..content2..'\n'
+       .. '   1     1    0 '..file3,
+       exec_capture('jumps'))
```

Comment:
there is also `:help getjumplist()`, would that be useful for checking this instead of a `:jumps` screentest?